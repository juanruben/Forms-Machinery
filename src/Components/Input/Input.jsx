import React from 'react';
import PropTypes from 'prop-types';
import './Input.scss';

const Input = (props) => {
    const {
        icon, errors, type, name, value, onChange, placeholder,
    } = props;
    const warning = (errors.length > 0);
    return (
        <div className="input-bordered-container">
            {icon && <i className={icon} />}
            <input className={`input-bordered ${warning && 'border-error'}`} type={type} name={name} value={value} onChange={onChange} placeholder={placeholder} />
            {warning && <i className="fas fa-exclamation warning" />}
        </div>
    );
};

Input.propTypes = {
    icon: PropTypes.string,
    errors: PropTypes.string,
    type: PropTypes.string,
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    value: PropTypes.node,
    placeholder: PropTypes.string,
};

Input.defaultProps = {
    icon: null,
    errors: '',
    type: 'text',
    value: '',
    placeholder: '',
};

export default Input;
