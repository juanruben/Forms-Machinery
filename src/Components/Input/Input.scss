@import "../../styles.scss";

.read-only-container {
  position: relative;

  i {
    color: rgb(134, 134, 134);
    position: absolute;
    left: 0px;
    top: 5px;
  }

  .padding-icon {
    padding-left: 20px;
  }
}

.input-bordered-container {
  position: relative;

  i {
    color: rgb(134, 134, 134);
    position: absolute;
    left: 14px;
    top: 12px;
  }

  input {
    border: solid $icafal-orange 1px;
    border-radius: 3px;
    display: block;
    height: 38px;
    margin-bottom: 10px;
    padding-left: 10px;
    width: 100%;

    &::placeholder {
      color: #bbb;
    }
  }

  .border-error {
    border: solid red 3px;
    border-radius: 4px;
  }

  .warning {
    color: red;
    position: absolute;
    left: calc(100% + 10px);
    top: 14px;
    width: 20px;
  }

  .padding-icon {
    padding-left: 40px;
  }
}

.bold-label {
  font-weight: bold;
}

.locked-value {
  padding-bottom: 20px;
  padding-left: 5px;
}
