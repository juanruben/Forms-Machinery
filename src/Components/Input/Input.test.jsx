import React from 'react';
import { shallow } from 'enzyme';
import Input from './Input';

describe('Input', () => {
    let wrapper;
    let callback;
    beforeEach(() => {
        callback = jest.fn();
        wrapper = shallow(<Input name="test" onChange={callback} />);
    });

    it('should render correctly', () => expect(wrapper.debug()).toMatchSnapshot());

    it('value check', () => {
        wrapper.find('input').simulate('change', { target: { value: 'test' } });
        expect(callback).toHaveBeenCalled();
    });

    it('should render text', () => {
        wrapper.setProps({ value: 'test' });
        expect(wrapper.find('input').prop('value')).toEqual('test');
    });

    it('should render icon', () => {
        wrapper.setProps({ icon: 'icon' });
        expect(wrapper.find('i')).toHaveLength(1);
    });

    it('should render error correctly', () => {
        wrapper.setProps({ errors: 'test' });
        expect(wrapper.find('i')).toHaveLength(1);
        expect(wrapper.find('input').hasClass('border-error')).toBe(true);
    });

    it('should render input', () => {
        expect(wrapper.find('input')).toHaveLength(1);
    });

    it('should render div', () => {
        expect(wrapper.find('div')).toHaveLength(1);
    });
});
