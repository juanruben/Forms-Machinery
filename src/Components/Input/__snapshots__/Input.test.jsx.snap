// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
"<div className=\\"input-bordered-container\\">
  <input className=\\"input-bordered false null\\" type=\\"text\\" name=\\"test\\" value=\\"\\" onChange={[Function: mockConstructor]} placeholder=\\"\\" />
</div>"
`;
